_posts/2016-08-03-testing-spark-streaming-integration-testing-with-docker-compose.md

```diff
@@ -0,0 +1,244 @@
+---
+layout: post
+asset-type: post
+name: Testing-Spark-Streaming-Integration-testing-with-Docker-Compose
+title: 'Testing Spark Streaming: Integration testing with Docker Compose'
+date: 2016-08-03 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/compose.jpg
+    attribution:
+      text: Untitled by Daida Medina
+      href: https://www.instagram.com/p/7Gjd9dHowa/?taken-by=thelastfreeusername
+tags:
+- testing
+- big data
+- spark
+- streaming
+
+---
+
+In the [first post]() of this series we saw how to unit test [Spark Streaming](http://spark.apache.org/streaming/) operations using [Spark Testing Base](https://github.com/holdenk/spark-testing-base). Here we'll see how to do integration testing using [Docker Compose](https://docs.docker.com/compose/).
+
+### What is Integration testing
+
+We [previously](http://robdmoore.id.au/blog/2015/01/26/testing-i-dont-even/) saw a discussion about unit and integration testing. Again, as we want to keep the post focused, we'll work with a definition of integration testing that holds these characteristics:
+
+- Network integration: our code should call the network to integrate with the third party dependencies. Part of our integration test efforts will be then verifying the behaviour of our code in the presence of network issues.
+
+- Framework integration: frameworks try to produce predictable and intuitive APIs. However, that's not always the case and integration testing give us verification about our assumptions.
+
+### What is Docker Compose
+
+[Docker](https://www.docker.com/what-docker) provides a lightweight and secure paradigm for virtualisation. As a consequence Docker is the perfect candidate to spin up and dispose container(processes) for integration testing. You can wrap your application or external dependencies in Docker containers and managing their lifecycle with ease.
+
+Orchestrating the relationships, order of execution or shared resources of a bunch of containers could be cumbersome and tedious. Instead of baking our own solutions with Bash scripts, we can use [Docker Compose](https://docs.docker.com/compose/).
+
+### Controlling the lifecycle
+
+Managing how and when a process should start, stop or move into different states is part of the process lifecycle management. Let's see some considerations about this management when integration testing.
+
+- If the process is expensive to spin up, from time or space point of view, keeping the process started for the whole test suite could be convenient.
```

Comment:
...keeping the process...

or

... maintaining the process...

Latter sounds better